ctpolicy/ctpolicy_test.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ctpolicy
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/google/certificate-transparency-go/loglist"
+	"github.com/google/certificate-transparency-go/testdata"
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/certificate-transparency-go/x509util"
+)
+
+func getTestCertPEMShort() *x509.Certificate {
+	cert, _ := x509util.CertificateFromPEM([]byte(testdata.TestCertPEM))
+	cert.NotAfter = time.Date(2013, 1, 1, 0, 0, 0, 0, time.UTC)
+	return cert
+}
+
+func getTestCertPEM2Years() *x509.Certificate {
+	cert, _ := x509util.CertificateFromPEM([]byte(testdata.TestCertPEM))
+	cert.NotAfter = time.Date(2014, 1, 1, 0, 0, 0, 0, time.UTC)
+	return cert
+}
+
+func getTestCertPEM3Years() *x509.Certificate {
+	cert, _ := x509util.CertificateFromPEM([]byte(testdata.TestCertPEM))
+	cert.NotAfter = time.Date(2015, 1, 1, 0, 0, 0, 0, time.UTC)
+	return cert
+}
+
+func getTestCertPEMLongOriginal() *x509.Certificate {
+	cert, _ := x509util.CertificateFromPEM([]byte(testdata.TestCertPEM))
+	return cert
+}
+
+func sampleLogList(t *testing.T) *loglist.LogList {
+	t.Helper()
+	var loglist loglist.LogList
+	err := json.Unmarshal([]byte(testdata.SampleLogList), &loglist)
+	if err != nil {
+		t.Error("Unable to Unmarshal testdata.SampleLogList.", err)
+		t.Fatal(err)
```

Comment:
Fold the `t.Error` together with the `t.Fatal`, and make it into a `t.Fatalf` (need to add `: %v` to the format string).

Reply:
What do you mean? For err formatting?

Reply:
Sorry, I could have been clearer.  I mean two things:
 1) If you do `t.Fatal`, there's no need to `t.Error` just beforehand - just use the `t.Error` message in the `t.Fatal` call.
 2) For the first invocation, you've got an `err` parameter but no `%v` in the format string and you're using `t.Error` not `t.Errorf`.

tl;dr: do (just) `t.Fatalf("Unable to Unmarshal testdata.SampleLogList: %v", err)`

Reply:
Done